lib/IDE/Formatting.cpp

```diff
@@ -2485,7 +2485,7 @@ class FormatWalker : public ASTWalker {
   getIndentContextFrom(CaptureListExpr *CL,
                        SourceLoc ContextLoc = SourceLoc()) {
     AbstractClosureExpr *CE = CL->getClosureBody();
-    BraceStmt *BS = CE->getBody();
+    BraceStmt *BS = CE ? CE->getBody() : NULL;
```

Comment:
None of these can ever be a null pointer, but the invariants are neglected. Can we assert this for the closure upon `CaptureListExpr` construction, and remove the always true `hasBody()` check from `AbstractClosureExpr::getBody`? This also seems to be the only remaining use of `hasBody()`, so we may as well just delete it.

Reply:
`lib/SIL/IR/SILProfiler.cpp` still uses `hasBody()` in assert. But if `hasBody()` always returns `true` this assert could also be removed.

Reply:
done

Reply:
> are you sure I should remove `hasBody()` completely?

At least I’d vote for it. @etcwilde What was the motivation for adding it in c57b80a3c88cb2eaf5825c0c22848f0d4ebab70c?